src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -2312,6 +2316,103 @@ public URL createCrumbedUrl(String relativePath) throws IOException {
         }
 
         /**
+         * Add the "Authorization" header with Basic credentials derived from login and password using Base64
+         * @since TODO
+         */
+        public @NonNull WebClient withBasicCredentials(@NonNull String login, @NonNull String passwordOrToken) {
+            addRequestHeader(HttpHeaders.AUTHORIZATION, "Basic " + Scrambler.scramble(login + ":" + passwordOrToken));
+            return this;
+        }
+
+        /**
+         * Add the "Authorization" header with Basic credentials derived from login and password using Base64
+         * After the closure being called, the header is removed
```

Comment:
Seems overly complicated. You can simply add a header to a single request. Or just call `createWebClient` again to get a fresh one without authentication, as we would already have done for `.login` calls.